### PR TITLE
Support environment variable expansion in host labels

### PR DESCRIPTION
**Pull request by ktsaou**
## Summary
- Allow `${VAR}` and `${VAR:-default}` syntax in `[host labels]` section of netdata.conf
- Variables are resolved from the environment at label load/reload time
- The raw `${VAR}` template is preserved in inicfg, so `netdata.conf` regeneration keeps the original syntax
- Unset variables with no default produce an empty value (sanitized to `[none]`) and log a warning

## Changes
- `src/database/rrdhost-labels.c` — added `env_expand_labels_value()` and `value_has_env_variables()`, modified `config_label_cb()` to expand before adding labels
- `src/database/rrdhost-labels.h` — declared `rrdhost_labels_unittest()`
- `src/daemon/main.c` — hooked unittest into `netdata -W unittest`
- `docs/netdata-agent/configuration/organize-systems-metrics-and-alerts.md` — documented the new syntax with examples

## Test plan
- [x] Unit tests: 38 test cases covering all corner cases (basic expansion, defaults, empty vars, malformed syntax, buffer overflow, no recursive expansion, detection function)
- [x] Build: compiles clean
- [x] `netdata -W unittest` passes
- [x] Manual: set env vars, configure labels with `${VAR}`, verify via `/api/v1/info`
- [x] Manual: test `reload-labels` picks up env changes
- [x] Manual: verify netdata.conf regeneration preserves `${VAR}` syntax

---
## Summary by cubic
Add environment variable expansion to [host labels] in netdata.conf for dynamic label values. Labels resolve on load and reload, so you can update them without restarting.

- **New Features**
  - Expand ${VAR} and ${VAR:-default} in label values; unset/empty without default resolve to [none] with a warning. Supports mixing with literals (e.g., ${DC}-${RACK}); no recursive expansion.
  - Preserve the original ${...} text in the config store so regenerated netdata.conf keeps placeholders.
  - Add unit tests (hooked to netdata -W unittest) and update docs with examples.

- **Migration**
  - Set env vars and use ${VAR} or ${VAR:-default} in [host labels].
  - Run netdatacli reload-labels to apply changes.

<sup>Written for commit 6135911c5b34b83c070ba9e9411c33cb195ef0ba. Summary will update on new commits.</sup>

### CODE REVIEW

File: docs/netdata-agent/configuration/organize-systems-metrics-and-alerts.md [hunk 1]
```diff
@@ -143,13 +143,30 @@ Add your own labels to categorize systems by any criteria you need.
...
+    [host labels]
+        region = ${REGION}
+        rack = ${RACK:-unknown}
+        env = ${DEPLOYMENT_ENV:-production}
+        location = ${DC}-${RACK:-default}
+    ```
+
+    | Syntax | Behavior |
+    |--------|----------|
+    | `${VAR}` | Replaced with the value of `VAR`. If unset or empty, the label value becomes `[none]` |
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply docs/netdata-agent/configuration/organize-systems-metrics-and-alerts.md:158]: this is wrong. empty values are set to `[none]` elsewhere
